### DISCUSSION TITLE
Tetris, young people, publicise and user base.

### DISCUSSION THREAD

**Discussion by wc-yaa** [VBAndCs/sVB-Small-Visual-Basic#61] [General]
Hi, I notice a new example which is Tetris. Well done. It is very nice to add this Tetris example to the package. Young people will be more interested in sVB if they discover this game example in the package. Sure. 

I remember a facebook group about BASIC. The group is really huge and there are over 20000 members. Nevertheless, the group is also weird. Almost all topics are about BASIC programming before 1990. Perhaps attracting young people to use BASIC is an important task of developer of each BASIC dialect.

**Comment by VBAndCs** [comment]
Thanks for this helpful info.

I am a member of that BASIC group, and it has more than 30,000 members now! But they are anti-Visual Basic and the being a part of the sVB name makes them crazy :D. Nut until now they only removed one of my posts  and I am not panned yet :D

I will look at Itch.io . I am always searching for BASIC communities, but they are always frustrating. Most of them are careless about the fate of the language and moved on to other languages like C# or Python. They only care about gaining money. While the others are old folks stuck with the past and refuse any change or any anything new!

My preferred group in the Small Basic Enthusiasts:

https://www.facebook.com/groups/58053367137

where I share what's new about sVB versions and any new sample I create with it. The group is not that active, but there are a couple of members that post interesting SB graphics.

Anyway, my main focus is on the kids and teens and their parents who wants to make them learn programming. And I hope the sVB enters the positive feedback, where new users talks about it and share new samples, so that they  bring in more new users and so on. In the last month, sVB downloads are more than those of the previous 10 month! But the total is still very low (a less more than 600 downloads only!)

The terries game is originally written by Small Basic (and published by the ID: tetris). It was perfectly running in early versions of sVB until I mad a massive breaking change by allowing local variables, so it stopped working, and after fixing the variable not defined errors, it runs but doesn't function correctly. I didn't have time to go through its 500 lines of code without a debugger, but after I added the debugger to sVB recently, I used it to fix it, and it is now up and running, and I improved its code in the tetris2 version by using timers.

You should try the debugger using these shortcut keys (until I add the toolbar buttons in sVB 3.0):

F9: Toggle breakpoint on the current code line

Ctrl+F5: run with debugging.

Shift+F5: stop debugging and end the running program.

F5: run without debugging, or continue running the debugger from a break mode.

F10: step over subroutine calls and stop at the next line in break mode, or run the debugger and stop at the first line.

Ctrl+F10: step over the current block and stop at the next line after the end of the block.

F11: step into subroutine calls to the next line in break mode, or run the debugger and stop at the first line.

Ctrl+F11: Pause the program and break on the current executing line. This works only if the program is running in debug mode.

F12: step out the current subroutine in break mode, or run the debugger and stop at the first line.

Ctrl+F12: step out the enclosing block and stop at the next line after the end of the block.

Regards.





________________________________
From: wc-yaa ***@***.***>
Sent: Saturday, April 6, 2024 10:07 AM
To: VBAndCs/sVB-Small-Visual-Basic ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [VBAndCs/sVB-Small-Visual-Basic] Tetris, young people, publicise and user base. (Discussion #61)


Hi, I notice a new example which is Tetris. Well done. It is very nice to add this Tetris example to the package. Young people will be more interested in sVB if they discover this game example in the package. Sure.

I remember a facebook group about BASIC. The group is really huge and there are over 20000 members. Nevertheless, the group is also weird. Almost all topics are about BASIC programming before 1990. Perhaps attracting young people to use BASIC is an important task of developer of each BASIC dialect.

Itch.io is quite well-known. Many youths and children like to visit Itch because many(free) web games and downloadable games are on Itch. On the other hand, a lot of BASIC dialect authors also use Itch to publicise their dialects. The information of BASIC dialects is on these authors' webpages.

Blitz3D ( blitzresearch.itch.io )
PlayBASIC ( https://playbasic.itch.io/ )
Craft Basic and Commando Basic( https://lucidapogee.itch.io/ )
AppGameKit ( https://thegamecreators.itch.io/ )
BlitzMax ( https://nitrologic.itch.io/ )
RCBasic ( https://n00bcode.itch.io/ )
MiniBASIC ( https://joestrout.itch.io/ )
BBC BASIC ( https://bbcbasic.itch.io/ )
ugBASIC ( https://spotlessmind1975.itch.io/ )
GB BASIC ( https://tonywang.itch.io/ )

Some are game languages: BlitzMax, etc. Some are general purpose languages which can still be used to write games: BBC BASIC, etc. Do you believe that Itch is so popular among modern BASIC dialect authors? Maybe the user base of sVB will be enlarged more quickly if a colourful and cool webpage is made for sVB on Itch. Anyway, I am only curious.

Perhaps the above information will inspire you to generate many other good ideas to promote sVB effecitively. Good luck.

**Reply by wc-yaa** [reply]
>I am a member of that BASIC group, and it has more than 30,000 members now! But they are anti-Visual Basic and the being a part of the sVB name makes them crazy :D. Nut until now they only removed one of my posts  and I am not panned yet :D

Wow, 30000. Amazing! 

Many members of that group may not be the target audience of sVB since they may be quite old. 

> I will look at Itch.io . 

Thanks for considering my idea again. Itch is a website which is similar to Youtube. Youtube is for videos and Itch is for games. An important thing is that many frequent visitors of Itch should be the target audience of sVB.

> My preferred group in the Small Basic Enthusiasts:

The group looks nice. Wow, somebody wrote there, "Well, I am 80 years old and I don't have the mental energy it would take to learn a new programming language. SVB looks like a real interesting program, but I got overwhelmed with the LD Extensions for SB." 

>  In the last month, sVB downloads are more than those of the previous 10 month! But the total is still very low (a less more than 600 downloads only!)

Around 600 last month? Not bad. Good job! 

I am not sure, if you put sVB on Itch, maybe number of downloads per month will be much larger than 600. Young people are more willing to try new things. 

> but after I added the debugger to sVB recently, I used it to fix it, and it is now up and running, and I improved its code in the tetris2 version by using timers.

Well done!!! sVB is really powerful now. The Tetris example can make good impression on the target audience of sVB. Sure. 

> You should try the debugger using these shortcut keys (until I add the toolbar buttons in sVB 3.0):

Thank you very much! Your guide is so detailed.
